### PR TITLE
remove `page_allocator` and reflector `Version`

### PR DESCRIPTION
**Pull request by jinzhongjia**
removed `page_allocator`, and use general purpose allocator

adn reflector the `version`, improved the readability of some codes 

By the way, multiple memory leaks occurred in versions using page_allocator in the past.

### CODE REVIEW

**Comment by jinzhongjia** [comment]
This can ensure small-grained memory application without having to apply for a whole page every time.

**Comment by jinzhongjia** [comment]
OMG, bro, why are you still using `c_allocator`?

**Comment by hendriknielaender** [comment]
> OMG, bro, why are you still using `c_allocator`?

That's a deprecated allocator 😅 when we still used libarchive for extracting. So good find, from your side.

**Comment by jinzhongjia** [comment]
The pr test also passed on Linux, and the segment fault error seemed to be fixed.
It can be merged currently

**Comment by jinzhongjia** [comment]
image

File: src/versions.zig [hunk 1]
```diff
@@ -1,42 +1,69 @@
 const std = @import("std");
 
-pub fn list(allocator: std.mem.Allocator) !std.ArrayList([]const u8) {
-    const url = "https://ziglang.org/download/index.json";
-    const uri = std.Uri.parse(url) catch unreachable;
+// TODO: The url should be stored in a separate config file
```
Comment (hendriknielaender) [inline src/versions.zig:3]: Please open an issue for this TODO.

**Review by hendriknielaender — APPROVED** [review]

**Comment by hendriknielaender** [comment]
I think we can also remove `linkLibC` from the build process.

**Pull request merged**